docs/e2e-test.md

```diff
@@ -0,0 +1,124 @@
+# APIs testing using HTTP files and Rest Client
+
+![testing-pyramid](./images/testing-pyramid.png)
+
+## Context
+
+During our recent engagement with a financial sector client, and after completing the initial iteration of the API endpoints implementation, an important task was to perform end-to-end testing of the API endpoints to ensure that the APIs are working as intended, and to enable developers to demonstrate their work at the end of each sprint. We compared various tools and chose HTTP files with HTTP/Rest Client for this purpose.
```

Comment:
```suggestion
In our recent engagement, we needed to test the API endpoints end-to-end each iteration of the implementation cycle. This would verify that the APIs functioned correctly and allow developers to demo their work after every sprint. We evaluated different tools and decided to use HTTP files with HTTP/Rest Client for this task.

```